c2corg_ui/views/area.py

```diff
@@ -0,0 +1,107 @@
+import logging
+
+from pyramid.renderers import render
+from pyramid.view import view_config
+
+from c2corg_ui.views.document import Document
+
+log = logging.getLogger(__name__)
+
+
+class Area(Document):
+
+    _API_ROUTE = 'areas'
+
+    @view_config(route_name='areas_index')
+    def index(self):
+        return self._index('c2corg_ui:templates/area/index.html')
+
+    @view_config(route_name='areas_sitemap',
+                 renderer='c2corg_ui:templates/area/sitemap.html')
+    @view_config(route_name='areas_sitemap_default',
+                 renderer='c2corg_ui:templates/area/sitemap.html')
+    def sitemap(self):
+        areas, total, filter_params, lang = self._get_documents()
+        self.template_input.update({
+            'areas': areas,
+            'total': total,
+            'filter_params': filter_params,
+            'lang': lang
+        })
+        return self.template_input
+
+    @view_config(route_name='areas_view_id')
+    @view_config(route_name='areas_view_id_lang')
+    def redirect_to_full_url(self):
+        self._redirect_to_full_url()
+
+    @view_config(route_name='areas_view')
+    def detail(self):
+        id, lang = self._validate_id_lang()
+
+        def render_page(area, locale):
+            # FIXME self._get_geometry(area['geometry']['geom_detail'])
+            # always returns None!?
```

Comment:
You mean ``area['geometry']['geom_detail']``?
It seems so:
```
geometry: {
version: 1,
geom: null,
geom_detail: "{"coordinates": [[[5012713.121726842, -1417626.9752307558], [5014720.969939139, -1419322.4914989625], [5016862.674699049, -1421910.3847504659], [5021369.178464449, -1419054.7784040233], [5025518.7314366605, -1425122.9418901356], [5025563.350285863, -1427398.5031975226], [5029177.477068069, -1426818.4581584025], [5036271.874084975, -1429807.9210522575], [5041358.422889586, -1431548.0561696014], [5043589.3653476825, -1434715.9944601902], [5041804.611381183, -1440918.0144938724], [5038547.435392276, -1439222.4982257595], [5037030.394520779, -1442524.293063814], [5033282.411191073, -1442568.9119129279], [5036271.874084975, -1447744.6984158468], [5032613.128453569, -1453500.529957872], [5034531.73896757, -1458944.0295557738], [5024314.022509259, -1461041.1154665023], [5018781.285213053, -1458542.4599133607], [5017041.150095645, -1454526.7634886259], [5019316.711402947, -1452965.1037679943], [5016238.010810747, -1447789.317264994], [5020253.707235457, -1447030.7968292553], [5022306.174296959, -1449886.403175674], [5024358.641358464, -1449217.1204382065], [5018647.4286655495, -1442702.7684604854], [5019673.662196248, -1440739.5390972751], [5019941.375291251, -1436902.3180692648], [5019048.998308056, -1434671.3756110768], [5016059.535414041, -1432529.67085128], [5013783.974106742, -1432038.8635104573], [5013159.3102184385, -1430298.7283930932], [5014185.543749245, -1429004.7817673117], [5012980.834821844, -1427844.6916890645], [5014364.01914584, -1425301.417286803], [5015211.77727994, -1424230.5649068854], [5013516.2610117365, -1421999.6224487284], [5010080.609626238, -1419411.7291973415], [5012713.121726842, -1417626.9752307558]]], "type": "Polygon"}"
```

Actually it can even be used "as is" in the mako templates: https://github.com/c2corg/v6_ui/blob/88704ca72061fa66d66729883985d3b3119ef82e/c2corg_ui/templates/area/view.html#L42
=> I wonder if we really need that ``self._get_geometry`` function here?!